src/managers/guilds.ts

```diff
@@ -48,5 +60,183 @@ export class GuildManager extends BaseManager<GuildPayload, Guild> {
       typeof template === 'object' ? template.code : template
     ].post({ name, icon })
     return new Guild(this.client, guild)
+
+  /**
+   * Creates a guild. Returns Guild. Fires guildCreate event.
+   * @param options Options for creating a guild
+   */
+  async create(options: GuildCreateOptions): Promise<Guild> {
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+    if (options.icon !== undefined && !options.icon.startsWith('data:')) {
+      options.icon = await fetchAuto(options.icon)
+    }
+
+    const body: GuildCreatePayload = {
+      name: options.name,
+      region: options.region,
+      icon: options.icon,
+      verification_level: options.verificationLevel,
+      roles:
+        options.roles !== undefined
+          ? options.roles.map((obj) => {
+              let result: GuildCreateRolePayload
+              if (obj instanceof Role) {
+                result = {
+                  id: obj.id,
+                  name: obj.name,
+                  color: obj.color,
+                  hoist: obj.hoist,
+                  position: obj.position,
+                  permissions: obj.permissions.bitfield.toString(),
+                  managed: obj.managed,
+                  mentionable: obj.mentionable
+                }
+              } else {
+                result = obj
+              }
+
+              return result
+            })
+          : undefined,
+      channels:
+        options.channels !== undefined
+          ? options.channels.map(
+              (obj): GuildCreateChannelPayload => ({
+                id: obj.id,
+                name: obj.name,
+                type: obj.type,
+                parent_id: obj.parentID
+              })
+            )
+          : undefined,
+      afk_channel_id: options.afkChannelID,
+      afk_timeout: options.afkTimeout,
+      system_channel_id: options.systemChannelID
+    }
+
+    const result: GuildPayload = await this.client.rest.post(GUILDS(), body)
+    const guild = new Guild(this.client, result)
+
+    return guild
```

Comment:
```javascript
return new Guild(this.client, result)
```

Reply:
what, did you read how is guild initialized